lib/ansible/plugins/lookup/sops.py

```diff
@@ -0,0 +1,154 @@
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: sops
+    author: Edoardo Tenani <e.tenani@arduino.cc>
+    version_added: "2.8"
+    short_description: Read sops encrypted file contents
+    description:
+        - This lookup returns the contents from a file on the Ansible controller's file system.
+        - This lookup requires the sops executable to be available in the controller PATH.
+    options:
+        _terms:
+            description: path(s) of files to read
+            required: True
+    notes:
+        - this lookup does not understand 'globing' - use the fileglob lookup instead.
+"""
+
+EXAMPLES = """
+tasks:
+  - name : output secrets to screen (BAD IDEA)
+    debug:
+        msg: "Content: {{lookup('sops', item)}}"
+    with_file:
+    - sops-encrypted-file.enc.yaml
+
+  - name: add ssh private key
+    copy:
+        content: "{{lookup('sops', user + '-id_rsa')}}"
+        dest: /home/{{user}}/.ssh/id_rsa
+        owner: "{{user}}"
+        group: "{{user}}"
+        mode: 0600
+"""
+
+RETURN = """
+    _raw:
+        description: decrypted file content
+"""
+
+# From https://github.com/mozilla/sops/blob/master/cmd/sops/codes/codes.go
+# Should be manually updated
+sops_error_codes = {
+    1: "ErrorGeneric",
+    2: "CouldNotReadInputFile",
+    3: "CouldNotWriteOutputFile",
+    4: "ErrorDumpingTree",
+    5: "ErrorReadingConfig",
+    6: "ErrorInvalidKMSEncryptionContextFormat",
+    7: "ErrorInvalidSetFormat",
+    8: "ErrorConflictingParameters",
+    21: "ErrorEncryptingMac",
+    23: "ErrorEncryptingTree",
+    24: "ErrorDecryptingMac",
+    25: "ErrorDecryptingTree",
+    49: "CannotChangeKeysFromNonExistentFile",
+    51: "MacMismatch",
+    52: "MacNotFound",
+    61: "ConfigFileNotFound",
+    85: "KeyboardInterrupt",
+    91: "InvalidTreePathFormat",
+    100: "NoFileSpecified",
+    128: "CouldNotRetrieveKey",
+    111: "NoEncryptionKeyFound",
+    200: "FileHasNotBeenModified",
+    201: "NoEditorFound",
+    202: "FailedToCompareVersions",
+    203: "FileAlreadyEncrypted"
+}
+
+
+class Sops():
+    @staticmethod
+    def decrypt(encrypted_file):
+        # Run sops directly, python module is deprecated
+        command = ["sops", "--decrypt", encrypted_file]
+        process = Popen(command, stdout=PIPE, stderr=PIPE)
+        (output, err) = process.communicate()
+        exit_code = process.wait()
+        return output, err, exit_code
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+        ret = []
+        for term in terms:
+            display.debug("Sops lookup term: %s" % term)
+            lookupfile = self.find_file_in_search_path(variables, 'files', term)
+            display.vvvv(u"Sops lookup using %s as file" % lookupfile)
+
+            try:
+                if lookupfile:
+                    # Run sops directly, python module is deprecated
+                    command = ["sops", "--decrypt", encrypted_file]
```

Comment:
This variable is causing an `undefined variable` error.
```suggestion
                    command = ["sops", "--decrypt", lookupfile]
```

Reply:
@endorama ping

Reply:
I pushed the fix!